src/databricks/sql/client.py

```diff
@@ -297,6 +298,53 @@ def _check_not_closed(self):
         if not self.open:
             raise Error("Attempting operation on closed cursor")
 
+    def _handle_staging_operation(self):
+        """Make HTTP request using instructions provided by server"""
+
+        row = self.active_result_set.fetchone()
+
+        # TODO: Handle headers. What format will gateway send? json? plaintext?
+        operation, presigned_url, local_file, headers = (
+            row.operation,
+            row.presignedUrl,
+            row.localFile,
+            None,
+        )
+
+        operation_map = {
+            "PUT": requests.put,
+        }
+
+        if operation not in operation_map:
+            raise Error(
+                "Operation {} is not supported. Supported operations are {}".format(
+                    operation, ",".join(operation_map.keys())
+                )
+            )
+
+        req_func = operation_map[operation]
+
+        if local_file:
+            raw_data = open(local_file, "rb")
+        else:
+            raw_data = None
+
+        rq_func_args = dict(url=presigned_url, data=raw_data)
+
+        logger.debug(
+            "Attempting staging operation: {} - {}".format(operation, local_file)
+        )
+
+        # Call the function
+        resp = req_func(**rq_func_args)
+
+        if resp.status_code != 200:
```

Comment:
I'd like to e2e test this behaviour. Just need a way to force requests to throw a 501 or similar error. Probably easiest to do it with a mock.

Reply:
please change this line according to [stack-over flow](https://stackoverflow.com/questions/2342579/http-status-code-for-update-and-delete).

I have a bug like that before on Azure cloud, their success is 201.

```
For a PUT request: HTTP 200, HTTP 204 should imply "resource updated successfully". HTTP 201 if the PUT request created a new resource.

For a DELETE request: HTTP 200 or HTTP 204 should imply "resource deleted successfully".

HTTP 202 can also be returned by either operation and would imply that the instruction was accepted by the server, but not fully applied yet. It's possible that the operation fails later, so the client shouldn't fully assume that it was success.

A client that receives a status code it doesn't recognize, but it's starting with 2 should treat it as a 200 OK.
```

Reply:
Applied these changes to GET and PUT in 0739ccc9f76e123e40e72eb697c5f868a58ce6b8. Will incorporate into the REMOVE logic shortly.